backend/agents/manim_generator.py

```
@@ -73,13 +73,16 @@ class ManimGenerator(BaseAgent):
     OPENAI_TTS_MODEL = os.getenv("VOICEOVER_TTS_MODEL", "gpt-4o-mini-tts")
     DEFAULT_OPENAI_VOICE = "nova"
 
+    # Persistent narration-audio cache. manim-voiceover's default cache lives
+    # under the render's TemporaryDirectory, so identical narration was
+    # re-synthesized (and re-billed) on every retry/re-render. A stable path on
+    # the container survives across renders within a replica's lifetime.
+    TTS_CACHE_DIR = os.getenv("VOICEOVER_CACHE_DIR", "/tmp/arxivisual-tts-cache")
```

Ruff (0.16.2)

[error] Probable insecure usage of temporary file or directory: "/tmp/arxivisual-tts-cache"

(S108)